File: brownfield-backend/src/main/java/com/petclinic/visit/VisitService.java
package com.petclinic.visit;

import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class VisitService {

    private final VisitRepository visitRepository;

    public VisitService(VisitRepository visitRepository) {
        this.visitRepository = visitRepository;
    }

    public List<Visit> findAll() {
        return visitRepository.findAll();
    }

    public Visit findById(Long id) {
        return visitRepository.findById(id).orElse(null);
    }

    public List<Visit> findByPetId(Long petId) {
        return visitRepository.findByPetId(petId);
    }
}

File: brownfield-backend/src/visit/visitService.ts
import { Visit } from "./visit";
import { VisitRepository } from "./visitRepository";

export class VisitService {
  private visitRepository: VisitRepository;

  constructor(visitRepository?: VisitRepository) {
    this.visitRepository = visitRepository ?? new VisitRepository();
  }

  findAll(): Promise<Visit[]> {
    return this.visitRepository.findAll();
  }

  findById(id: number): Promise<Visit | null> {
    return this.visitRepository.findById(id);
  }

  findByPetId(petId: number): Promise<Visit[]> {
    return this.visitRepository.findByPetId(petId);
  }
}
